### ISSUE TITLE
support python 3.9

### ISSUE THREAD

**Issue by cyberic99** [nir/jupylet#26]
Hi,

When trying to install jupylet on archlinux, with python 3.9.2, I get:

```shell
$ pip install jupylet --user --verbose
```
I get:
```python
$ pip install jupylet --user --verbose
Using pip 20.2.3 from /home/eric/.local/lib/python3.9/site-packages/pip (python 3.9)
User install by explicit request
Created temporary directory: /tmp/pip-ephem-wheel-cache-3wvo_4kp
Created temporary directory: /tmp/pip-req-tracker-_m4zqxh6
Initialized build tracking at /tmp/pip-req-tracker-_m4zqxh6
Created build tracker: /tmp/pip-req-tracker-_m4zqxh6
Entered build tracker: /tmp/pip-req-tracker-_m4zqxh6
Created temporary directory: /tmp/pip-install-ui97ta1b
1 location(s) to search for versions of jupylet:
* https://pypi.org/simple/jupylet/
Fetching project page and analyzing links: https://pypi.org/simple/jupylet/
Getting page https://pypi.org/simple/jupylet/
Found index url https://pypi.org/simple
Looking up "https://pypi.org/simple/jupylet/" in the cache
Request header has "max_age" as 0, cache bypassed
Starting new HTTPS connection (1): pypi.org:443
https://pypi.org:443 "GET /simple/jupylet/ HTTP/1.1" 304 0
  Link requires a different Python (3.9.2 not in: '>=3.6, <3.9'): https://files.pythonhosted.org/packages/04/43/73317ed5ac537cc113f15ba8961a82b0becf58dd38d208b8aa53f6c6cd71/jupylet-0.8.2.tar.gz#sha256=65b29691461836ff56800596f74f4bf10620c7734982babe7e44c145bebef712 (from https://pypi.org/simple/jupylet/) (requires-python:>=3.6, <3.9)
  Link requires a different Python (3.9.2 not in: '>=3.6, <3.9'): https://files.pythonhosted.org/packages/d5/4a/759e69ce9edb0faf295efd9852354c810ba2d126f9d01d666694ab56a91e/jupylet-0.8.3.tar.gz#sha256=d8587839e785976ede70b54c2281044d3c6ab3ef99655c46d25259807cad4205 (from https://pypi.org/simple/jupylet/) (requires-python:>=3.6, <3.9)
  Link requires a different Python (3.9.2 not in: '>=3.6, <3.9'): https://files.pythonhosted.org/packages/39/ce/b57320a8f4338dddc64a17065b9c61c8a8513a309043bda3d8731152bc66/jupylet-0.8.4.tar.gz#sha256=796637f1bca3e1d63d2838a69f4cd4bd7f4e449f210feb2e7360c4ce20bcc85c (from https://pypi.org/simple/jupylet/) (requires-python:>=3.6, <3.9)
  Link requires a different Python (3.9.2 not in: '>=3.6, <3.9'): https://files.pythonhosted.org/packages/a0/71/b99740ec81e8e5450e3a2c40ee439a765c86f1f63eedc9e76d0c54f66fa7/jupylet-0.8.5.tar.gz#sha256=3a02ce9816cd7dd9067c7712befe0f8b4416819dc52771f9718230d1f8641d2a (from https://pypi.org/simple/jupylet/) (requires-python:>=3.6, <3.9)
  Link requires a different Python (3.9.2 not in: '>=3.6, <3.9'): https://files.pythonhosted.org/packages/c7/40/97e4e17afb76dda53df66f8b7d96794e3eb8b628195bc224b9df1985e6ac/jupylet-0.8.6.tar.gz#sha256=5b803c8dcbd7b69998cc24380c44d1764de2071df728206f9c6bc18a62a7cca6 (from https://pypi.org/simple/jupylet/) (requires-python:>=3.6, <3.9)
  Link requires a different Python (3.9.2 not in: '>=3.6, <3.9'): https://files.pythonhosted.org/packages/56/66/14403cfa25c6cc6d44d16d1b4b1e644a22315e4b9de5bce4cab8f3e7fcdb/jupylet-0.8.7.tar.gz#sha256=9b3323acd71db6e5df3e684ed66f2fb1d8c267331b2fc2b4dddcae973c0c86a9 (from https://pypi.org/simple/jupylet/) (requires-python:>=3.6, <3.9)
Given no hashes to check 0 links for project 'jupylet': discarding no candidates
ERROR: Could not find a version that satisfies the requirement jupylet (from versions: none)
ERROR: No matching distribution found for jupylet
Exception information:
Traceback (most recent call last):
  File "/home/eric/.local/lib/python3.9/site-packages/pip/_internal/cli/base_command.py", line 228, in _main
    status = self.run(options, args)
  File "/home/eric/.local/lib/python3.9/site-packages/pip/_internal/cli/req_command.py", line 182, in wrapper
    return func(self, options, args)
  File "/home/eric/.local/lib/python3.9/site-packages/pip/_internal/commands/install.py", line 323, in run
    requirement_set = resolver.resolve(
  File "/home/eric/.local/lib/python3.9/site-packages/pip/_internal/resolution/legacy/resolver.py", line 183, in resolve
    discovered_reqs.extend(self._resolve_one(requirement_set, req))
  File "/home/eric/.local/lib/python3.9/site-packages/pip/_internal/resolution/legacy/resolver.py", line 388, in _resolve_one
    abstract_dist = self._get_abstract_dist_for(req_to_install)
  File "/home/eric/.local/lib/python3.9/site-packages/pip/_internal/resolution/legacy/resolver.py", line 339, in _get_abstract_dist_for
    self._populate_link(req)
  File "/home/eric/.local/lib/python3.9/site-packages/pip/_internal/resolution/legacy/resolver.py", line 305, in _populate_link
    req.link = self._find_requirement_link(req)
  File "/home/eric/.local/lib/python3.9/site-packages/pip/_internal/resolution/legacy/resolver.py", line 270, in _find_requirement_link
    best_candidate = self.finder.find_requirement(req, upgrade)
  File "/home/eric/.local/lib/python3.9/site-packages/pip/_internal/index/package_finder.py", line 926, in find_requirement
    raise DistributionNotFound(
pip._internal.exceptions.DistributionNotFound: No matching distribution found for jupylet
WARNING: You are using pip version 20.2.3; however, version 21.0.1 is available.
You should consider upgrading via the '/usr/bin/python3 -m pip install --upgrade pip' command.
Removed build tracker: '/tmp/pip-req-tracker-_m4zqxh6'
```
Is there anything I can do about it?

Thank you

**Comment by nir** [comment]
Jupylet does not install on python 3.9 since some of its dependencies do not have pre-compiled wheels for that version and are therefore difficult to install (e.g. on Windows).

If you know what you are doing and have an environment setup properly with build tools, you can try to clone the repository and install with pip -e <PATH_TO_JUPYLET> to circumvent the version check.

**Comment by cyberic99** [comment]
Thank you, that did the trick.

I tested some shaders from shadertoys, and everything seemed to work on python 3.9.2.

**Comment by nir** [comment]
It appears that most of the missing dependencies have released a wheel for python 3.9 by now, so I will check if I can create a new release for python 3.9 this weekend.

Thanks for opening this issue! :)

**Comment by nir** [comment]
Python 3.9 supported in version 0.8.8
